nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/SplitPCAP.java

```diff
@@ -0,0 +1,244 @@
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("If a FlowFile cannot be transformed from the configured input format to the configured output format, "
+            + "the unchanged FlowFile will be routed to this relationship.")
+            .build();
+    public static final Relationship REL_SPLIT = new Relationship.Builder()
+            .name("split")
+            .description("The individual PCAP 'segments' of the original PCAP FlowFile will be routed to this relationship.")
+            .build();
+
+    private static final List<PropertyDescriptor> DESCRIPTORS = List.of(PCAP_MAX_SIZE);
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_ORIGINAL, REL_FAILURE, REL_SPLIT);
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    /**
+     * This method is called when a trigger event occurs in the processor.
+     * It processes the incoming flow file, splits it into smaller pcap files based on the PCAP Max Size,
+     * and transfers the split pcap files to the success relationship.
+     * If the flow file is empty or not parseable, it is transferred to the failure relationship.
+     *
+     * @param context  the process context
+     * @param session  the process session
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final List<FlowFile> splitFilesList = new ArrayList<>();
+
+        try {
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream inStream) {
+                    try {
+                        final int pcapMaxSize = context.getProperty(PCAP_MAX_SIZE.getName()).asDataSize(DataUnit.B).intValue();
+                        final List<Packet> loadedPackets = new ArrayList<>();
+                        final BufferedInputStream bInStream = new BufferedInputStream(inStream);
+
+                        if ( bInStream.available() == 0 ) {
+                            throw new IOException("PCAP file empty.");
+                        }
+
+                        final byte[] pcapHeaderArray = new byte[PCAP.PCAP_HEADER_LENGTH];
+                        bInStream.read(pcapHeaderArray, 0, PCAP.PCAP_HEADER_LENGTH);
+                        int currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+
+                        final PCAP templatePcap = new PCAP(new ByteBufferInterface(pcapHeaderArray));
+
+                        while (bInStream.available() > 0) {
+
+                            byte[] packetHeaderArray = new byte[Packet.PACKET_HEADER_LENGTH];
```

Comment:
Not necessary to have `List`  for variable names. A plural is fine to indicate a collection. In addition no need to have an `Array` suffix as `pcapHeader` and `packetHeader` are descriptive enough.

```suggestion
        final List<FlowFile> splitFiles = new ArrayList<>();

        try {
            session.read(flowFile, new InputStreamCallback() {
                @Override
                public void process(final InputStream inStream) {
                    try {
                        final int pcapMaxSize = context.getProperty(PCAP_MAX_SIZE.getName()).asDataSize(DataUnit.B).intValue();
                        final List<Packet> loadedPackets = new ArrayList<>();
                        final BufferedInputStream bInStream = new BufferedInputStream(inStream);

                        if ( bInStream.available() == 0 ) {
                            throw new IOException("PCAP file empty.");
                        }

                        final byte[] pcapHeader = new byte[PCAP.PCAP_HEADER_LENGTH];
                        bInStream.read(pcapHeader, 0, PCAP.PCAP_HEADER_LENGTH);
                        int currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;

                        final PCAP templatePcap = new PCAP(new ByteBufferInterface(pcapHeader));

                        while (bInStream.available() > 0) {

                            byte[] packetHeader = new byte[Packet.PACKET_HEADER_LENGTH];
```